[Feat] Redis 추가

Galmaeki opened the pull request:
# [Feature] Redis 의존성 등록
- 이메일 인증의 TTL 설정의 용이함과 캐싱 등의 기능을 이용하기 위해 Redis 의존성을 추가 했습니다

## 변경 사항
### [fix] build.gradle 의존성 추가
변경사항 :
- Spring Boot Starter Data Redis 의존성 추가

### [fix] application.yml
변경사항 :
- application.yml에 각각 redis 연결 설정을 추가했습니다

### [feat] dockercompose-local.yml 추가
변경사항 :
- 로컬에서 DB의 편리한 사용을 위해 MySQL과 Redis 컨테이너 설정을 작성해둔 파일입니다

## 관련 이슈

- #43 

## 개발 유형

- [ ] 버그 수정
- [x] 새로운 기능 개발
- [ ] 리팩토링
- [ ] 테스트 코드 작성
- [ ] 문서 업데이트
- 기타 (아래에 설명 추가)

## 작업 기간

- 작업 시작일: (2024-05-09)
- 작업 종료일: (2024-05-09)

## 유의사항
- 

## 참고문헌

Galmaeki committed:
[feat] Redis 의존성 추가
- build.gradle data-redis 의존성 추가
- application.yml : 개발환경, 로컬환경 redis 추가

Galmaeki committed:
[feat] 개발 편의용 로컬 DockerConpose 추가
- dockercompose-local.yml : 로컬에서 사용하기 위한 DockerCompose 파일 추가

Galmaeki committed:
[feat] DockerConpose MYSQL 추가
- dockercompose-local.yml : 로컬 실행 용도의 MySQL 컨테이너 설정 추가

ghrltjdtprbs commented on src/main/resources/application-develop.yml:
```diff
@@ -11,3 +11,10 @@ spring:
       ddl-auto: update
+
+#  cache:
+#    type: redis
```
불필요한 주석은 삭제하는게 더 깔끔할 것 같습니다!😊

ghrltjdtprbs commented on dockercompose-local.yml:
로컬 환경에서도 도커를 사용할 계획이시군요! 
그렇다면 mysql의 비밀번호 또한 application-local.yml 파일과 동일하게 .env 파일을 통해 개개인 로컬 환경에 맞춰 변경할 수 있도록 하는게 좋을 것 같습니다! 일관성을 유지하면서도 보안측면에서도 좋을 것 같아요👍

ghrltjdtprbs reviewed (approved):
수고하셨습니다! 몇가지 코멘트 남겼으니 참고해주세요 ㅎㅎ 😊
참고로 application-local파일의 환경 변수를 .env로 관리하는 것처럼 dockercompose-local파일의 환경 변수 또한 .env 파일로 관리가 가능합니다. 걱정하시는 redis url, port 노출 문제는 .env로 해결이 가능할 것 같습니다

Galmaeki committed:
[fix] Redis 설정 .env 사용하도록 변경
- appplication-local : .env 활용하도록 코드 수정
- .env.sample : 샘플 코드 수정

Galmaeki committed:
[fix] Redis Develop 설정 변경
- appplication-develop : dockerfile에서 주입받도록 변경
- Dockerfile : redis 설정 주입받도록 수정
- ci-cd.yml : secrets 주입받도록 수정

Galmaeki committed:
[fix] 불필요한 주석 삭제
- appplication-develop : 불필요한 주석 삭제

Galmaeki committed:
[fix] dockercompose-local 기본 포트 수정
- appplication-develop : 기본 포트 3300 -> 3306으로 수정

Galmaeki committed:
[fix] dockercompose-local .env 사용하도록 변경
- dockercompose-local.yml : .env 사용하도록 수정

asd42270 reviewed (approved):
고생하셨습니다!!
